src/stores/ProofStore.ts

```diff
@@ -1,126 +1,72 @@
-      this.proofsCompleted.filter((proof) => proof.dataType === ledgerName)
+  equal(a: Proof, b: Proof): boolean {
+    return (
+      a.origin === b.origin &&
+      a.account === b.account &&
+      a.dataType === b.dataType
     )
   }
 
-  get emailProofsCompleted() {
-    return this.proofsCompleted.filter(
-      (proof) => proof instanceof EmailProof
-    ) as EmailProof[]
+  deleteProof(a: Proof) {
+    this.proofsCompleted = this.proofsCompleted.filter((b) => !this.equal(a, b))
   }
 
-  get eRC721ProofsCompleted() {
-    return this.proofsCompleted.filter(
-      (proof) => proof instanceof ERC721Proof
-    ) as ERC721Proof[]
-  }
-
-  async generateEmail(domain: string, signature: string) {
-    try {
-      // Get public key
-      const eddsaPublicKey = await getEddsaPublicKey()
-      // Check navigator availability
-      checkNavigator()
-      // Create proof
-      const newEmailProof = new EmailProof(domain)
-      await newEmailProof.build(signature, eddsaPublicKey)
-      this.proofsCompleted.push(newEmailProof)
-      return newEmailProof
-    } catch (e) {
-      handleError(e)
+  deleteIfUsed(error: unknown, proof: Proof) {
+    if (
+      proof &&
+      error instanceof Error &&
+      error.message.includes('This ZK proof has already been used')
+    ) {
+      this.deleteProof(proof)
     }
   }
 
-  async generateERC721(contract: string, network: Network) {
+  async mint(proof: Proof) {
     try {
-      // Get the account
-      const account = WalletStore.account
-      if (!account) {
-        throw new Error('No account selected')
-      }
-      // Get public key
-      const eddsaPublicKey = await getEddsaPublicKey()
-      // Get nullifier signature
-      const nullifierMessage = getNullifierMessage()
-      const nullifierSignature = await WalletStore.signMessage(nullifierMessage)
-      // Get ownership EdDSA attestation
-      const ownershipSignature = await requestAddressOwnershipAttestation(
-        nullifierSignature,
-        nullifierMessage
-      )
-      // Get balance EdDSA attestation
-      const balanceSignature = await requestBalanceAttestation(
-        contract,
-        network,
-        account
-      )
-      // Get the proof
-      const newERC721Proof = new ERC721Proof(contract, account, network)
-      checkNavigator()
-      await newERC721Proof.build(
-        ownershipSignature,
-        balanceSignature,
-        eddsaPublicKey
-      )
-      this.proofsCompleted.push(newERC721Proof)
-      return newERC721Proof
-    } catch (e) {
-      handleError(e)
+      const provider = await WalletStore.createGSNProvider()
+      const result = await this.createBadge(provider, proof)
+      this.deleteProof(proof)
+      return result
+    } catch (error) {
+      this.deleteIfUsed(error, proof)
+      handleError(error)
+      throw error
     }
   }
 }
 
-export default proxy(new ProofStore()).makePersistent(env.VITE_ENCRYPT_KEY)
+export default proxy(
```

Comment:
Do we have `proxy(proxy())` here? Why do we need two proxies here? Maybe we need to explain this line in the comment?

Reply:
We organize the combination of proxies